### PR TITLE
[Feature] Support YOLOv6 ML model

### PR DESCRIPTION
**Pull request by wanghonglie**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Suport ML model for YOLOv6

## Modification

- New layers:    BepC3StageBlock
- New backbone:  YOLOv6CSPBep
- New Neck:      YOLOv6CSPRepPAFPN

### CODE REVIEW

File: configs/yolov6/README.md [hunk 1]
```diff
@@ -16,11 +16,13 @@ For years, YOLO series have been de facto industry-level standard for efficient
...
+| YOLOv6-m |  P5  | 640  |  300  |  Yes   | Yes |  16.69   |  48.4  | [config](../yolov6/yolov6_m_syncbn_fast_8xb32-400e_coco.py) |                                                                                                                                                   [model](<>) \| [log](<>)                                                                                                                                                   |
+| YOLOv6-l |  P5  | 640  |  300  |  Yes   | Yes |  20.86   |  51.0  | [config](../yolov6/yolov6_l_syncbn_fast_8xb32-300e_coco.py) |                                                                                                                                                   [model](<>) \| [log](<>)                                                                                                                                                   |
 
 **Note**:
```
Comment (hhaAndroid) [inline configs/yolov6/README.md:27]: 下面说一下：我们观察到官方最近发布了 base 模型，虽然精度有一些下降但是更高效，后续我们也将提供 base 模型配置

File: configs/yolov6/yolov6_l_syncbn_fast_8xb32-300e_coco.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        type='YOLOv6CSPRepPAFPN',
+        deepen_factor=deepen_factor,
+        widen_factor=widen_factor,
+        expansion=1. / 2,
+        block_cfg=dict(
+            type='ConvWrapper',
+            norm_cfg=dict(type='BN', momentum=0.03, eps=0.001)),
+        csp_act_cfg=dict(type='SiLU', inplace=True)),
+    bbox_head=dict(
+        type='YOLOv6Head',
```
Comment (hhaAndroid) [inline configs/yolov6/yolov6_l_syncbn_fast_8xb32-300e_coco.py:26]: 删掉

File: configs/yolov6/yolov6_l_syncbn_fast_8xb32-300e_coco.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+_base_ = './yolov6_m_syncbn_fast_8xb32-300e_coco.py'
+
+deepen_factor = 1
+widen_factor = 1
+
+model = dict(
+    backbone=dict(
+        type='YOLOv6CSPBep',
```
Comment (hhaAndroid) [inline configs/yolov6/yolov6_l_syncbn_fast_8xb32-300e_coco.py:8]: 一样的参数就不要写了

File: configs/yolov6/yolov6_m_syncbn_fast_8xb32-300e_coco.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    neck=dict(
+        type='YOLOv6CSPRepPAFPN',
+        deepen_factor=deepen_factor,
+        widen_factor=widen_factor,
+        block_cfg=dict(type='RepVGGBlock'),
+        expansion=2. / 3,
+        csp_act_cfg=dict(type='ReLU', inplace=True)),
+    bbox_head=dict(
+        type='YOLOv6Head',
+        head_module=dict(widen_factor=widen_factor),
```
Comment (hhaAndroid) [inline configs/yolov6/yolov6_m_syncbn_fast_8xb32-300e_coco.py:24]:
```suggestion
        head_module=dict(widen_factor=widen_factor)
```

File: configs/yolov6/yolov6_l_syncbn_fast_8xb32-300e_coco.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        deepen_factor=deepen_factor,
+        widen_factor=widen_factor,
+        expansion=1. / 2,
+        block_cfg=dict(
```
Comment (hhaAndroid) [inline configs/yolov6/yolov6_l_syncbn_fast_8xb32-300e_coco.py:12]: block_cfg 名字会不会太大了

**Comment by PeterH0323** [comment]
Pls fix the lint

File: mmyolo/models/layers/__init__.py [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
                           RepStageBlock, RepVGGBlock, SPPFBottleneck,
                           SPPFCSPBlock)
 
 __all__ = [
     'SPPFBottleneck', 'RepVGGBlock', 'RepStageBlock', 'ExpMomentumEMA',
     'ELANBlock', 'MaxPoolAndStrideConvBlock', 'SPPFCSPBlock',
-    'PPYOLOEBasicBlock', 'EffectiveSELayer'
-]
+    'PPYOLOEBasicBlock', 'EffectiveSELayer', 'BepC3StageBlock'
+]
```
Comment (PeterH0323) [inline mmyolo/models/layers/__init__.py:12]:
```suggestion
]

```

File: mmyolo/models/layers/yolo_bricks.py [hunk 7]
```diff
@@ -348,38 +348,145 @@ def switch_to_deploy(self):
...
+                 n=1,
+                 expansion=0.5,
+                 concat=True,
+                 block_cfg=dict(type='RepVGGBlock'),
+                 norm_cfg: ConfigType = dict(
+                     type='BN', momentum=0.03, eps=0.001),
+                 act_cfg: ConfigType = dict(type='ReLU', inplace=True)
+                 ):  # ch_in, ch_out, number, shortcut, groups, expansion
+        super().__init__()
+        c_ = int(out_channels * expansion)  # hidden channels
```
Comment (PeterH0323) [inline mmyolo/models/layers/yolo_bricks.py:367]:
```suggestion
        hidden_channels = int(out_channels * expansion)  # hidden channels
```

File: mmyolo/models/layers/yolo_bricks.py [hunk 8]
```diff
@@ -348,38 +348,145 @@ def switch_to_deploy(self):
...
+
+        self.cv1 = ConvModule(
```
Comment (PeterH0323) [inline mmyolo/models/layers/yolo_bricks.py:369]:
```suggestion
        self.conv1 = ConvModule(
```

Comment (PeterH0323) [reply mmyolo/models/layers/yolo_bricks.py:369]: 下面的 cv2 cv3 m 也要改一改

File: tests/test_models/test_backbone/test_efficient_rep.py [hunk 9]
```diff
@@ -111,3 +111,92 @@ def test_forward(self):
...
+        model = YOLOv6CSPBep(frozen_stages=frozen_stages)
+        model.init_weights()
+        model.train()
+
+        for mod in model.stem.modules():
+            for param in mod.parameters():
+                assert param.requires_grad is False
+        for i in range(1, frozen_stages + 1):
+            layer = getattr(model, f'stage{i}')
+            for mod in layer.modules():
```
Comment (PeterH0323) [inline tests/test_models/test_backbone/test_efficient_rep.py:127]:
```suggestion
            for layer_module in layer.modules():
```

**Pull request closed without merging**